sqlglot/optimizer/qualify_columns.py

```diff
@@ -213,6 +214,31 @@ def _qualify_columns(scope, resolver):
             # column_table can be a '' because bigquery unnest has no table alias
             if column_table:
                 column.set("table", column_table)
+        elif column_table not in scope.sources:
+            # structs are used like tables (e.g. "struct"."field"), so they need to be qualified
+            # separately by parsing them as <table>.<column>.dot.dot.dot ...
+
+            struct_fields = [val for val in reversed(list(column.args.values())) if val is not None]
```

Comment:
do you need to call list() ?

Reply:
Python 3.7 doesn't support `reversed` for `dict_values`, so I opted for this workaround for the time being.